### PR TITLE
add prometheus integration.

### PR DESCRIPTION
**Pull request by JaredTan95**
support https://github.com/ctripcorp/apollo/issues/2489

- Exposed prometheus API: `/prometheus`  in Apollo-ConfigService, Apollo-Adminservice,Apollo-Portal modules.
  e.g.
  - localhost:8080/prometheus
  - localhost:8090/prometheus
  - localhost:8070/prometheus

- How to use: https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#production-ready-metrics-export-prometheus

### CODE REVIEW

File: apollo-common/src/main/java/com/ctrip/framework/apollo/common/controller/WebMvcConfig.java [hunk 1]
```diff
@@ -29,7 +27,7 @@ public void addArgumentResolvers(List<HandlerMethodArgumentResolver> argumentRes
   @Override
   public void configureContentNegotiation(ContentNegotiationConfigurer configurer) {
     configurer.favorPathExtension(false);
-    configurer.ignoreAcceptHeader(true).defaultContentType(MediaType.APPLICATION_JSON_UTF8);
+    //configurer.ignoreAcceptHeader(true).defaultContentType(MediaType.APPLICATION_JSON_UTF8);
```
Comment (nobodyiam) [inline apollo-common/src/main/java/com/ctrip/framework/apollo/common/controller/WebMvcConfig.java]: It's better to remove it if it's not necessary.

File: apollo-common/src/main/resources/application.properties [hunk 2]
```diff
@@ -17,3 +17,5 @@ server.max-http-header-size=10240
 management.endpoints.web.base-path=/
 logging.file.max-size=50MB
 logging.file.max-history=10
+
+management.endpoints.web.exposure.include=*
```
Comment (nobodyiam) [inline apollo-common/src/main/resources/application.properties]: It is dangerous to expose all the endpoints, the default is `info, health`, so how about change it to `info,health, prometheus `

File: pom.xml [hunk 3]
```diff
@@ -362,6 +362,21 @@
 				</exclusion>
 			</exclusions>
 		</dependency>
+
+		<!-- Spring boot actuator to expose metrics endpoint -->
+		<dependency>
```
Comment (nobodyiam) [inline pom.xml]: These dependencies are only needed in config-service/admin-service and portal, so they should be moved to `apollo-common`.

Besides, I think they also caused the travis-ci build failure for apollo-core module.

File: apollo-common/pom.xml [hunk 4]
```diff
@@ -60,5 +60,20 @@
             <groupId>commons-lang</groupId>
             <artifactId>commons-lang</artifactId>
         </dependency>
+
+        <!-- Spring boot actuator to expose metrics endpoint -->
+        <dependency>
```
Comment (nobodyiam) [inline apollo-common/pom.xml]: it looks like `spring-boot-starter-actuator` was already delcared in line 22...

File: apollo-common/pom.xml [hunk 5]
```diff
@@ -60,5 +60,20 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-actuator</artifactId>
+        </dependency>
+        <!-- Micormeter core dependecy -->
```
Comment (nobodyiam) [inline apollo-common/pom.xml]: Micormeter => Micrometer ?

File: apollo-common/src/main/resources/application.properties [hunk 6]
```diff
@@ -17,3 +17,7 @@ server.max-http-header-size=10240
 management.endpoints.web.base-path=/
 logging.file.max-size=50MB
 logging.file.max-history=10
+
+management.endpoints.web.exposure.include=info,health,metrics,prometheus
+management.endpoint.metrics.enabled=true
```
Comment (nobodyiam) [inline apollo-common/src/main/resources/application.properties]: According to [Enabling Endpoints](https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-endpoints.html#production-ready-endpoints-enabling-endpoints): by default, all endpoints except for shutdown are enabled. So it's not necessary to set metrics and prometheus endpoints enabled?

Comment (JaredTan95) [reply apollo-common/src/main/resources/application.properties]: No need indeed, just to show other developers. Or delete it.

File: apollo-common/pom.xml (hunk 5 above)
Comment (JaredTan95) [reply apollo-common/pom.xml]: yeah, typo

**Pull request merged**